### PR TITLE
Use cleaner warp_output directory structure

### PR DESCRIPTION
**Pull request by JorikSchellekens**
This changes the output from `<solfile-root>__WC__<contract-name>.cairo` to
`solfile.sol/contract.cairo` which is similar to how artifacts are generated in
hardhat and is much cleaner

### CODE REVIEW

File: src/passes/sourceUnitSplitter.ts [hunk 1]
```diff
@@ -113,11 +105,11 @@ function updateScope(nodes: readonly Scoped[], newScope: number): readonly Scope
 }
 
 export function mangleFreeFilePath(path: string): string {
-  return `${path}${FREE_FILE_SUFFIX}`;
+  return `${path}/${FREE_FILE_NAME}`;
```
Comment (swapnilraj) [inline src/passes/sourceUnitSplitter.ts:108]: path.resolve to be compatible with all OSs

File: src/testing.ts [hunk 2]
```diff
@@ -349,7 +348,7 @@ function combineResults(results: ResultType[]): ResultType {
 function getTestsWithUnexpectedResults(results: Map<string, ResultType>): string[] {
   const testsWithUnexpectedResults: string[] = [];
   const groupedResults = groupBy([...results.entries()], ([file, _]) => {
-    return file.split(CONTRACT_INFIX)[0];
+    return file.split('/').slice(0, -1).join('/');
```
Comment (swapnilraj) [inline src/testing.ts:351]: path.basename

File: src/testing.ts [hunk 3]
```diff
@@ -381,16 +380,12 @@ function printResults(results: Map<string, ResultType>, unexpectedResults: strin
       console.log(`Actual outcome:`);
       const Actual = new Map<string, ResultType>();
       results.forEach((value, key) => {
-        if (key === o || key.startsWith(`${o}${CONTRACT_INFIX}`)) {
+        if (key === o || key.startsWith(`${o}`)) {
```
Comment (swapnilraj) [inline src/testing.ts:383]: just key.starsWith(o) will work as well

File: src/testing.ts (hunk 2 above)
Comment (JorikSchellekens) [reply src/testing.ts:351]: dirname*

File: src/testing.ts [hunk 4]
```diff
@@ -34,216 +34,219 @@ const ResultTypeOrder = [
...
+    ['example_contracts/using_for/library.sol', 'Success'],
+    ['example_contracts/using_for/simple.sol', 'Success'],
+    ['example_contracts/usingReturnValues.sol', 'Success'],
+    ['example_contracts/userDefinedFunctionCalls.sol', 'Success'],
+    ['example_contracts/userdefinedtypes.sol', 'Success'],
+    ['example_contracts/userdefinedidentifier.sol', 'Success'],
+    ['example_contracts/variable_declarations.sol', 'Success'],
+    ['example_contracts/view_function.sol', 'Success'],
+    ['example_contracts/typestrings/enumArrays.sol', 'Success'],
+    //uses 'this.sol' keyword in the constructor
```
Comment (swapnilraj) [inline src/testing.ts:240]: your macro caught this as well

File: tests/behaviour/expectations/types.ts [hunk 5]
```diff
@@ -16,10 +16,16 @@ export class AsyncTest {
     return `${this.name}.sol`;
   }
   get cairo() {
-    return `warp_output/${mangleContractFilePath(manglePath(this.name), this.contract)}.cairo`;
+    return `warp_output/${mangleContractFilePath(
```
Comment (swapnilraj) [inline tests/behaviour/expectations/types.ts:19]: resolve here as well

**Comment by swapnilraj** [comment]
this will needed to be fixed
https://github.com/NethermindEth/UniStark/blob/main/test/shared/utilities.ts#L52

File: src/testing.ts (hunk 4 above)
Comment (JorikSchellekens) [reply src/testing.ts:240]:
```qc :'( ESC
100 @c```

**Comment by JorikSchellekens** [comment]
https://github.com/NethermindEth/UniStark/pull/37

**Comment by JorikSchellekens** [comment]
Final result:
```
 ± : tree artifacts                                                   ⏎ [3h21m]
artifacts
└── contracts
    ├── UniswapV3Factory.sol
    │   ├── UniswapV3Factory.cairo
    │   ├── UniswapV3Factory_abi.json
    │   └── UniswapV3Factory_compiled.json
    ├── UniswapV3Pool.sol
    │   ├── UniswapV3Pool.cairo
    │   ├── UniswapV3Pool_abi.json
    │   └── UniswapV3Pool_compiled.json
    ├── UniswapV3PoolDeployer.sol
    │   ├── UniswapV3PoolDeployer.cairo
    │   ├── UniswapV3PoolDeployer_abi.json
    │   └── UniswapV3PoolDeployer_compiled.json
    └── test
        ├── BitMathEchidnaTest.sol
        │   ├── BitMathEchidnaTest.cairo
        │   ├── BitMathEchidnaTest_abi.json
        │   └── BitMathEchidnaTest_compiled.json
        ├── BitMathTest.sol
        │   ├── BitMathTest.cairo
        │   ├── BitMathTest_abi.json
        │   └── BitMathTest_compiled.json
        ├── FullMathEchidnaTest.sol
        │   ├── FullMathEchidnaTest.cairo
        │   ├── FullMathEchidnaTest_abi.json
        │   └── FullMathEchidnaTest_compiled.json

```

**Comment by JorikSchellekens** [comment]
I'm very happy with this change

**Review by swapnilraj — APPROVED** [review]

**Comment by swapnilraj** [comment]
Yep this looks mucho better

**Comment by swapnilraj** [comment]
@GuyMcComb unleash the semantic tests on this pls

**Review by rodrigo-pino — APPROVED** [review]

**Comment by GuyMcComb** [comment]
Sem Tests Passed

**Pull request merged**